markets/perps-market/contracts/storage/PerpsMarket.sol

```diff
@@ -30,6 +30,15 @@ library PerpsMarket {
 
     error PriceFeedNotSet(uint128 marketId);
 
+    event MarketUpdated(
```

Comment:
would not recommend adding events to libraries.  libraries are more reusable code and should be used in conjunection with a contract.  you can then define your event and emit it within contract.

you'll also notice there's no events in any other storage libraries for reason above.

Reply:
my recommendation would be to create a new interface called: `IMarketEvents.sol`:
```
interface IMarketEvents {
    event MarketUpdated(...);
}


```

Then in `AsyncOrderModule`, you can inteherit from this interface like so:
```
contract AsyncOrderModule is IAsyncOrderModule, IMarketEvents 
```

and now you can do:

```
emit MarketUpdated(...);
```

This way in liquidations, I can also inherit from `IMarketEvents` and emit the same event

Reply:
My idea was that instead of having to remember to `emit MarketUpdates` in both the order and the liquidation module they both can now call ``Perps.updatePositionData` and the event is emitted for them.

But agree that is' a little weird to emit event from the library, so if you think manually emitting the same event twice is cleaner I'm happy to move to a IMarketEvents architecture

Reply:
yeah understood.  i think its ok to return the relevant data to the caller and let the module decide if it wants to emit an event. thoughts?

Reply:
Yep, will do. Just explained my thinking

Reply:
I would probably leave a comment that caller should emit this though. If you call this and forget emit an event offchain/subgraph tracking will be wrong

Reply:
yeah, nothing contract side breaks but yeah, in general, we'll audit all events together with troy at offsite maybe

Reply:
Done